src/Microsoft.AspNetCore.Sockets.Client/Connection.cs

```diff
@@ -14,70 +14,174 @@ namespace Microsoft.AspNetCore.Sockets.Client
 {
     public class Connection : IDisposable
     {
+        private readonly ILoggerFactory _loggerFactory;
+        private readonly ILogger _logger;
+        private int _connectionState = ConnectionState.Initial;
         private IChannelConnection<Message> _transportChannel;
         private ITransport _transport;
-        private readonly ILogger _logger;
+        private Task _receiveLoopTask;
+
+        private ReadableChannel<Message> Input => _transportChannel.Input;
+        private WritableChannel<Message> Output => _transportChannel.Output;
 
         public Uri Url { get; }
 
-        private Connection(Uri url, ITransport transport, IChannelConnection<Message> transportChannel, ILogger logger)
+        public event Action Connected;
+        public event Action<byte[], Format> Received;
+        public event Action<Exception> Closed;
+
+        public Connection(Uri url)
+            : this(url, null)
+        { }
+
+        public Connection(Uri url, ILoggerFactory loggerFactory)
         {
-            Url = url;
+            Url = url ?? throw new ArgumentNullException(nameof(url));
 
-            _logger = logger;
-            _transport = transport;
-            _transportChannel = transportChannel;
+            _loggerFactory = loggerFactory ?? NullLoggerFactory.Instance;
+            _logger = _loggerFactory.CreateLogger<Connection>();
         }
 
-        private ReadableChannel<Message> Input => _transportChannel.Input;
-        private WritableChannel<Message> Output => _transportChannel.Output;
+        public Task StartAsync() => StartAsync(null, null);
```

Comment:
I prefer when Intellisense basically Shows `StartAsync()` instead of overwhelming you with `StartAsync(ITransport transport = null, HttpClient httpClient = null)` - especially for most common functions.